bubblewrap.c

```diff
@@ -1388,23 +1418,52 @@ main (int    argc,
       die_with_error ("Creating new namespace failed");
     }
 
+  ns_uid = opt_sandbox_uid;
+  ns_gid = opt_sandbox_gid;
+
   if (pid != 0)
     {
+      if (is_privileged && opt_unshare_user)
+        {
+          /* Map the uid/gid 0 if opt_needs_devpts, as otherwise
+           * mounting it will fail.
+           * Due to this non-direct mapping we need to have set[ug]id
+           * caps in the parent namespaces, and thus we need to write
+           * the map in the parent namespace, not the child. */
+          write_uid_gid_map (ns_uid, uid,
+                             ns_gid, gid,
+                             pid, TRUE, opt_needs_devpts);
+        }
+
       /* Initial launched process, wait for exec:ed command to exit */
 
       /* We don't need any caps in the launcher, drop them immediately. */
       drop_caps ();
+
+      /* Let child run */
+      val = 1;
+      res = write (child_wait_fd, &val, 8);
```

Comment:
The idiom I normally use for functions where ignoring the result is a warning, e.g. the `write()` in dbus-daemon's pipe-to-self, is a block containing only comments:

```
if (write (child_wait_fd, &val, 8) < 8)
  {
    /* Ignore result, if e.g. the child died and closed ... */
  }
```